extensions/wikia/PortableInfobox/services/Helpers/PortableInfoboxTemplatesHelper.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Wikia\PortableInfobox\Helpers;
+
+/**
+ * Helper used for checking if template consist any hidden infoboxes (eg. put into <includeonly> tag)
+ *
+ * Class PortableInfoboxTemplatesHelper
+ */
+class PortableInfoboxTemplatesHelper {
+
+	/**
+	 * @desc For given Title, get property 'infoboxes' from parser output. If property is empty, this may mean that
+	 * template is inside the <noinclude> tag. In this case, we want to skip the <includeonly> tags, get from this only
+	 * infoboxes and parse them again to check their presence and get params.
+	 *
+	 * @param $title \Title
+	 *
+	 * @return mixed false when no infoboxes found, Array with infoboxes on success
+	 */
+	public function parseInfoboxes( $title ) {
```

Comment:
This method looks out of sync with its description. Please also consider splitting it into smaller chunks (i.e. the logic within if ($includeonlyText) block and infobox cleanup loop )